File: evd_ros_backend/evd_ros_core/src/evd_script/program/tasks/simple_pick_and_place.py
from ..task import Task
from ..primitives import MoveTrajectory
from .open_gripper import OpenGripper
from .close_gripper import CloseGripper


class SimplePickAndPlace(Task):

    '''
    Data structure methods
    '''

    @classmethod
    def type_string(cls):
        return 'simple-pick-and-place.'

    @classmethod
    def full_type_string(cls):
        return Task.full_type_string() + cls.type_string()

    def __init__(self, startLocUuid=None, pickLocUuid=None, placeLocUuid=None, thing_uuid=None,
                 type='', name='', uuid=None, parent=None, append_type=True, primitives=None):

        if primitives == None:
            primitives = [
                MoveTrajectory(startLocUuid,pickLocUuid),
                CloseGripper(thing_uuid=thing_uuid),
                MoveTrajectory(pickLocUuid,placeLocUuid),
                OpenGripper(thing_uuid=thing_uuid)
            ]

        super(SimplePickAndPlace,self).__init__(
            type='simple-pick-and-place.'+type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type,
            primitives=primitives)

File: evd_authoring_app/src/model/evd_script/program/tasks/simple_pick_and_place.js
import { MoveTrajectory } from '../primitives';
import { CloseGripper } from './close_gripper';
import { OpenGripper } from './open_gripper';
import { Task } from '../Task';


export class SimplePickAndPlace extends Task {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'simple-pick-and-place';
    }

    static fullTypeString() {
        return Task.fullTypeString() + SimplePickAndPlace.typeString();
    }

    constructor(startLocUuid=null, pickLocUuid=null, placeLocUuid=null, thingUuid=null, type='', name='', uuid=null, parent=null, appendType=true, primitives=null) {

        if (primitives === null) {
            primitives = [
                MoveTrajectory(startLocUuid,pickLocUuid),
                CloseGripper(thingUuid=thingUuid),
                MoveTrajectory(pickLocUuid,placeLocUuid),
                OpenGripper(thingUuid=thingUuid)
            ];
        }

        super(
            type= (appendType) ? 'simple-pick-and-place.'+type : type,
            name= name,
            uuid= uuid,
            parent= parent,
            appendType= appendType,
            primitives= primitives
        );
    }

    static fromDict(dct) {
        return SimplePickAndPlace(
            type= dct.type,
            name= dct.name,
            uuid= dct.uuid,
            appendType= false,
            primitives= dct.primitives.map(p => NodeParser(p))
        );
    }
}